### PR TITLE
add prow job for istio upgrade e2e tests

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/17950

### CODE REVIEW

File: prow/upgrade-tests.sh [hunk 1]
```diff
@@ -25,42 +27,39 @@ WD=$(dirname "$0")
...
-UPGRADE_TEST_LOCAL="${UPGRADE_TEST_LOCAL:-}"
-
-# This is a script to download release artifacts from monthly or daily release
-# location and kick off upgrade/downgrade tests.
-#
-# This is currently triggered by https://github.com/istio-releases/daily-release
-# for release qualification.
-#
-# Expects HUB, SOURCE_VERSION, TARGET_VERSION, SOURCE_RELEASE_PATH, and TARGET_RELEASE_PATH as inputs.
+UPGRADE_TEST_LOCAL="${UPGRADE_TEST_LOCAL:-"true"}"
```
Comment (howardjohn) [inline prow/upgrade-tests.sh]: I think you probably either need this or kind. We want kind at some point but right now looks like we have nothing

Comment (carolynhu) [reply prow/upgrade-tests.sh]: yes, later on we will use kind

**Pull request merged**